tests/test_router_bgp.rb

```diff
@@ -540,31 +538,32 @@ def test_routerbgp_set_get_confederation_peers
       # confederation id be configured first so the expectation
       # in the next test is an empty peer list
       bgp.confederation_id = 55
+
       assert_empty(bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be empty")
-      bgp.confederation_peers_set(15)
+      bgp.confederation_peers = (15)
       assert_equal('15', bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be '15'")
-      bgp.confederation_peers_set(16)
+      bgp.confederation_peers = (16)
       assert_equal('16', bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be '16'")
-      bgp.confederation_peers_set(55.77)
+      bgp.confederation_peers = (55.77)
       assert_equal('55.77', bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be" \
+                   "vrf #{vrf}: bgp confederation_peers list should be " \
                    "'55.77'")
-      bgp.confederation_peers_set('15 16 55.77 18 555 299')
-      assert_equal('15 16 55.77 18 555 299',
+      bgp.confederation_peers = ('15 16 55.77 18 555 299')
```

Comment:
if you go by my above suggestion about changing this parameter to a proper array, this would be:
`bgp.confederation_peers = ['15', '16', '55.77', '18', '555', '299']`